week-1-ts-ecosystem/w01e03-object-oriented-structure/src/Cart.ts

```diff
@@ -0,0 +1,47 @@
+import { Product } from './Products.js'
+
+export class Cart<ProductType extends Product> {
+    private products: Array<ProductType>
+
+    constructor (initialProducts: ProductType[] = []) {
+        this.products = initialProducts
+    }
+
+    addProduct (product: ProductType): void {
+        this.products.push(product)
+    }
+
+    getProduct (id: Product['id']): ProductType | undefined {
+        return this.products.find(product => product.id === id)
+    }
+
+    getProducts (): ProductType[] {
+        return this.products
+    }
+
+    updateProduct (id: Product['id'], newProduct: ProductType): void {
+        const index = this.products.findIndex(product => product.id === id)
+        if (index === -1) {
+            return
+        }
+        this.products[index] = newProduct
+    }
+
+    removeProduct (id: Product['id']): void {
+        const index = this.products.findIndex(product => product.id === id)
```

Comment:
Ten fragment logiki powtarza Ci się w 2 miejscach, może warto to zrobić jako `private` wrzucić metodę 

`const index = this.getProductIndex(id)`

zwracać `index` i wykorzystać go w `removeProduct` , `updateProduct` oraz `getProduct`